lib/ansible/modules/packaging/os/package_facts.py

```diff
@@ -139,95 +150,275 @@
           }
         }
 '''
+import json
+
+from abc import ABCMeta, abstractmethod
+
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.basic import AnsibleModule, get_all_subclasses
+from ansible.module_utils.six import with_metaclass
+
+
+class TestFailed(Exception):
+    pass
+
+
+class PkgMgr(with_metaclass(ABCMeta, object)):
+
```

Comment:
Base should probably have a standardized attribute used in child classes:
```suggestion

    CLI = None

```

Reply:
I'd change them to abstractproperty:
``` python
@abstractattribute
def CLI(self):
    # This should simply be an attribute containing the UNIX command for this package manager.
    pass
```

This way any class which implements the interface will need to override it with a CLI attribute or property.

Reply:
Agreed. That's even better!